Jenkinsfile

```diff
@@ -1,81 +1,107 @@
-@Library('xmos_jenkins_shared_library@v0.32.0') _
+@Library('xmos_jenkins_shared_library@develop') _
+
+def clone_test_deps() {
+  dir("${WORKSPACE}") {
+    sh "ls ."
+    sh "git clone git@github.com:xmos/test_support"
+  }
+}
 
 getApproval()
 
 pipeline {
   agent none
+  environment {
+    REPO = 'lib_i2c'
+  }
+  options {
+    buildDiscarder(xmosDiscardBuildSettings())
+    skipDefaultCheckout()
+    timestamps()
+  }
+  parameters {
+    string(
+      name: 'TOOLS_VERSION',
+      defaultValue: '15.3.0',
+      description: 'The XTC tools version'
+    )
+    string(
+      name: 'XMOSDOC_VERSION',
+      defaultValue: 'v6.0.0',
+      description: 'The xmosdoc version')
+  }
   stages {
-    stage('Standard build and XS3 tests') {
+    stage('Build and test') {
       agent {
-        label 'x86_64&&macOS'
-      }
-      environment {
-        REPO = 'lib_i2c'
-        VIEW = getViewName(REPO)
-      }
-      options {
-        skipDefaultCheckout()
+        label 'x86_64 && linux'
       }
       stages {
-        stage('Get view') {
+        stage('Build examples') {
           steps {
-            xcorePrepareSandbox("${VIEW}", "${REPO}")
-          }
-        }
-        stage('Library checks') {
-          steps {
-             xcoreLibraryChecks("${REPO}")
-          }
-        }
-        stage('xCORE App XS2 builds') {
-          steps {
-            forAllMatch("${REPO}/examples", "app_*/") { path ->
-              runXmake(path)
-            }
-            forAllMatch("${REPO}/examples", "AN*/") { path ->
-              runXmake(path)
+            println "Stage running on ${env.NODE_NAME}"
+
+            dir("${REPO}") {
+              checkout scm
+
+              dir("examples") {
+                withTools(params.TOOLS_VERSION) {
+                  sh 'cmake -G "Unix Makefiles" -B build'
+                  sh 'xmake -C build -j 8'
```

Comment:
I think the main executors have 16 cores / 32 threads.

Reply:
I'm leaving it as 8 to match the template in https://xmosjira.atlassian.net/wiki/spaces/UAAI/pages/4233101313/XCommon+CMake+in+USB+Audio+Libraries#Jenkinsfile-template